app/models/container_deployment.rb

```diff
@@ -6,5 +6,395 @@ class ContainerDeployment < ApplicationRecord
+                    container_deployment_node.vm.hardware.ipaddresses.last + (!container_deployment_node.labels.empty? && add_labels ? " openshift_node_labels=#{container_deployment_node.labels}" : "")
+                  elsif container_deployment_node.address
+                    container_deployment_node.address + (!container_deployment_node.labels.empty? && add_labels ? " openshift_node_labels=#{container_deployment_node.labels}" : "")
+                  end
+    end
+    ip_array
+  end
+
+  def add_deployment_provider(options)
+    provider = ExtManagementSystem.new(
+      :type      => options[:provider_type],
+      :name      => options[:provider_name],
+      :port      => options[:provider_port],
+      :hostname  => options[:provider_hostname],
+      :ipaddress => options[:provider_ipaddress])
+    provider.save!
+    provider.update_authentication(:bearer => {:auth_key => options[:auth_key], :save => true})
+    valid_provider = provider.authentication_check.first
+    if valid_provider
+      self.deployed_ems = provider
+      save!
+    else
+      provider.destroy
+    end
+    valid_provider
+  end
+
+  def assign_container_deployment_node(options)
+    send(options[:type]).each do |deployment_node|
+      next unless deployment_node.vm.nil?
+      deployment_node.vm = Vm.find(options[:vm_id])
+      deployment_node.save!
+    end
+  end
+
+  # will be removed once moving to yaml format, no need to review method
+  def generate_ansible_inventory_for_subscription
+    template = <<eos
+[OSEv3:children]
+masters
+nodes
+
+[OSEv3:vars]
+ansible_ssh_user=#{ssh_auth.userid}
+deployment_type=openshift-enterprise
+rhsub_user=#{rhsm_auth.rhsm_user}
+rhsub_pass=#{rhsm_auth.password_encrypted}
+rhsub_pool=#{rhsm_auth.rhsm_sku}
+
+[masters]
+#{masters_addresses.join("\n") unless masters.empty?}
+
+[nodes]
+#{nodes_addresses.join("\n") unless nodes.empty?}
+eos
+    template
+  end
+
+  # will be removed once moving to yaml format, no need to review method
+  def generate_ansible_inventory
+    template = <<eos
+[OSEv3:vars]
+#{identity_provider_auth.first.generate_ansible_entry}
+#{"containerized=true" if containerized}
+[OSEv3:children]
+masters
+nodes
+
+[masters:vars]
+ansible_ssh_user=#{ssh_auth.userid}
+ansible_sudo=true
+deployment_type=#{kind}
+#openshift_use_manageiq=True
+
+[nodes:vars]
+ansible_ssh_user=#{ssh_auth.userid}
+ansible_sudo=true
+deployment_type=#{kind}
+#openshift_use_manageiq=True
+#{add_role("nfs") unless nfs_addresses.empty?}
+
+[masters]
+#{(deployment_master_address.to_s + " ansible_connection=local openshift_scheduleable=True") unless deployment_master_address.nil?}
+    #{masters_addresses.join("\n") unless masters.empty?}
+[nodes]
+#{nodes_addresses.join("\n") unless nodes.empty?}
+eos
+    template
+  end
+
+  def generate_ansible_yaml
+    hash = {}
+    hash[:version] = version
+    hash[:deployment] = {}
+    hash[:deployment][:ansible_config] = "/usr/share/atomic-openshift-utils/ansible.cfg"
```

Comment:
Is it a good idea to have `"/usr/share/atomic-openshift-utils/ansible.cfg"` hard-coded here?